src/java/org/apache/cassandra/net/InboundSink.java

```diff
@@ -96,8 +98,10 @@ public void accept(Message<?> message)
         {
             fail(message.header, t);
 
-            if (t instanceof TombstoneOverwhelmingException || t instanceof IndexNotAvailableException)
-                noSpamLogger.error(t.getMessage());
```

Comment:
@jonmeredith What if we just extended `NoSpamLogger` to take a properly typed `Throwable` and pass that to the underlying `Logger`? It would avoid the lambda creation at least...